plugins/inputs/powerdns_recursor/protocol_commons.go

```diff
@@ -0,0 +1,96 @@
+package powerdns_recursor
+
+import (
+	"encoding/binary"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"unsafe"
+)
+
+func parseResponse(metrics string) map[string]interface{} {
+	values := make(map[string]interface{})
+
+	s := strings.Split(metrics, "\n")
+
+	for _, metric := range s[:len(s)-1] {
+		m := strings.Split(metric, "\t")
+
+		if len(m) < 2 {
+			continue
+		}
+
+		i, err := strconv.ParseInt(m[1], 10, 64)
+
+		if err != nil {
+			continue
+		}
```

Comment:
```suggestion
		i, err := strconv.ParseInt(m[1], 10, 64)
		if err != nil {
			continue
		}
```

Reply:
so for now we are tossing anything that isn't an int, are there cases of floats for example?

Reply:
No. It looks like for this command everything is an integer of varying width (up to 64 bits), converted to a string.